spec/compiler/compiler_spec.cr

```diff
@@ -28,6 +28,24 @@ describe "Compiler" do
     end
   end
 
+  it "outputs correct cc command in preference to a filename when cross compile" do
+    crystal_bin = File.realpath(ENV["CRYSTAL_SPEC_COMPILER_BIN"]? || "bin/crystal")
+    Dir.cd compiler_datapath do
+      with_temp_executable "compiler_spec_output" do |path|
+        FileUtils.mkdir_p("#{path}/bin")
+        expected_bin = "#{path}/bin/compiler_sample-linux-x86_64"
+        output = IO::Memory.new
+        Process.run(
```

Comment:
This is effectively an integration spec and will fail if a local Crystal compiler isn't built, as the spec will simply pick up the existing Crystal installation. You have two choices here:

* If you could get hold of a `Crystal::Compiler` in this spec, it has a `#stdout` property which you could reassign the above `IO::Memory` to. Then you could call `Crystal::Command.run` directly, which uses the fresh compiler code, and test the compiler output as usual. The easiest way would probably be adding a new `stdout` parameter in `Command#initialize` and using it in `#create_compiler`.
* Move this spec into a separate test suite, say `spec/compiler/integration_spec.cr`, and create a corresponding new Makefile target that depends on the fresh compiler, similar to how `$(O)/primitives_spec` depends on `$(O)/crystal`. All the CI workflows must be updated to run this. This spec alone should be fast to build (i.e. it never requires the compiler source).

Reply:
This is true, that spec can also fail in the same way

Reply:
For the record, I think a unit test would be best for this. It may even need some refactoring to make that easier.
But we really just need a method that produces the linker command based on some pathnames and a basically empty `Program` instance with some flags configured.